comp/core/healthprobe/impl/healthprobe_test.go

```diff
@@ -13,33 +13,33 @@ import (
 	"net/http/httptest"
 	"testing"
 
-	healthprobeComponent "github.com/DataDog/datadog-agent/comp/core/healthprobe"
+	healthprobeComponent "github.com/DataDog/datadog-agent/comp/core/healthprobe/def"
 	"github.com/DataDog/datadog-agent/comp/core/log"
 	"github.com/DataDog/datadog-agent/comp/core/log/logimpl"
+	compdef "github.com/DataDog/datadog-agent/comp/def"
 	"github.com/DataDog/datadog-agent/pkg/status/health"
 	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
 	"github.com/stretchr/testify/assert"
-	"go.uber.org/fx"
-	"go.uber.org/fx/fxtest"
 )
 
 func TestServer(t *testing.T) {
-	lc := fxtest.NewLifecycle(t)
-
-	probe, err := newHealthProbe(lc,
-		fxutil.Test[dependencies](
-			t,
-			logimpl.MockModule(),
-			fx.Provide(func() healthprobeComponent.Options {
-				return healthprobeComponent.Options{
-					Port: 7869,
-				}
-			}),
-		),
-	)
+
+	lc := compdef.NewTestLifecycle()
+	logComponent := fxutil.Test[log.Component](t, logimpl.MockModule())
+
+	requires := Requires{
+		Lc:  lc,
+		Log: logComponent,
+		Options: healthprobeComponent.Options{
+			Port: 7869,
+		},
+	}
+
+	provides, err := NewComponent(requires)
```

Comment:
Did you consider creating the component using fxutil.Test?

Reply:
We want to avoid the use of Fx inside the `impl` folder + the impl code does not uses Fx

Reply:
Also, one of the benefit for this new file structure is that we no longer have to dela with Fx for testing components logic.